### PR TITLE
Fix for duplicate webhook policy name

### PR DESCRIPTION
**Pull request by tedgxt**
Signed-off-by: guanxiatao <guanxiatao@corp.netease.com>

### CODE REVIEW

File: make/migrations/postgresql/0031_2.0.1_schema.up.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/*remove the constraint for name in table 'notification_policy'*/
+ALTER TABLE notification_policy DROP CONSTRAINT notification_policy_name_key;
```
Comment (ywk253100) [inline make/migrations/postgresql/0031_2.0.1_schema.up.sql:2]: You should add the sql into the sql file for 2.1: https://github.com/goharbor/harbor/blob/master/make/migrations/postgresql/0040_2.1.0_schema.up.sql

Comment (tedgxt) [reply make/migrations/postgresql/0031_2.0.1_schema.up.sql:2]: fixed

File: src/core/api/notification_policy.go [hunk 2]
```diff
@@ -128,6 +129,11 @@ func (w *NotificationPolicyAPI) Post() {
 
 	id, err := notification.PolicyMgr.Create(policy)
 	if err != nil {
+		if e := orm.AsConflictError(err, "notification policy named %s already exists", policy.Name); e != nil {
```
Comment (wy65701436) [inline src/core/api/notification_policy.go:132]: we should put the logic in the dao level, like
https://github.com/goharbor/harbor/blob/master/src/pkg/label/dao.go#L79

Comment (tedgxt) [reply src/core/api/notification_policy.go:132]: considering the 409 / 500 response code should be sent in different circumstances

Comment (heww) [reply src/core/api/notification_policy.go:132]: You can use the `SendError` method which will return the response code depending on the err.

Comment (tedgxt) [reply src/core/api/notification_policy.go]: fixed

File: src/core/api/notification_policy.go [hunk 3]
```diff
@@ -180,6 +181,11 @@ func (w *NotificationPolicyAPI) Put() {
 	policy.ProjectID = w.project.ProjectID
 
 	if err = notification.PolicyMgr.Update(policy); err != nil {
+		if e := orm.AsConflictError(err, "notification policy named %s already exists", policy.Name); e != nil {
```
Comment (ywk253100) [inline src/core/api/notification_policy.go:184]: Why don't we call `SendError()` directly here?

Comment (heww) [reply src/core/api/notification_policy.go:184]: I think he wants to return special error messages for different errors. But I found that the error message in this handler may be duplicated with the error message in dao. Eg, `failed to update the notification policy: failed to update the notification policy: balabala balabala`

Comment (tedgxt) [reply src/core/api/notification_policy.go]: already fixed

**Review by wy65701436 — APPROVED** [review]
lgtm

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**